lib/active_record/connection_adapters/postgis/type.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Type
+    ##
+    # Attributes are looked up based on the type and the current adapter.
+    # For example, attribute :foo, :string has a type :string.
+    #
+    # The issue is that our adapter is :postgis, but all of the
+    # attributes are registered under :postgresql. This overwrite
+    # just forces us to always return :postgresql.
+    #
+    # The caveat is that when registering spatial types, we have
+    # to specify :postgesql as the adapter, not :postgis.
+    #
+    # ex. ActiveRecord::Type.register(:st_point, OID::Spatial, adapter: :postgresql)
+    class << self
+      def adapter_name_from(_model)
+        :postgresql
+      end
+
+      private
+
+      def current_adapter_name
+        :postgresql
+      end
+    end
+  end
+end
```

Comment:
I really do not grab that one : why do we need to monkey patch `ActiveRecord::Type` rather than just what you did in the next file, registering with correct type `postgresql` ? It feels like this should be under `ConnectionAdapters::PostGIS`

Reply:
The reason we can't just register the types under `postgresql` is that the `ActiveRecord::Type` module returns `:postgis` for `adapter_name_from`. This means that we miss out on all of the `postgresql` specific types (https://github.com/rails/rails/blob/ee7cf8cf7569ef87079c48ee81c867eae5e24ed4/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L948-L967) even though they're compatible with PostGIS.

One option could just be to re-register those above types and leave our name as `:postgis` in the type module, but then we would have to make sure we're up to date with the types every time there is an update to ActiveRecord (not that big of a deal, but still an additional thing to check).

Does that answer your questions?

Reply:
Partially : in the example you gave me we can see that postgresql is not guessed from any method. And what is bothering me is that we monkey patch some active_record common stuff, not under any namespace. Hence I am worried that we may have conflict with other gems.. But there may still be something I don't fully understand...

Reply:
Yeah, that makes sense. Here's the original `adapter_from_name` method (https://github.com/rails/rails/blob/beed933b9bb9f317697a95a36da067a67797e0d8/activerecord/lib/active_record/type.rb#L49-L51), and here are all the places it is used (https://github.com/rails/rails/search?q=adapter_name_from), so we can see it's pretty limited in its usage. 

That being said, you're right it's probably not ideal to be modifying the ActiveRecord internals since other gems could be relying on it being consistent with the adapter name. I think the best solution is to just copy the postgres specific types and register them under `:postgis` in this adapter.